WebCrawler/Model/LinkParser.cs

```diff
@@ -0,0 +1,195 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Text.RegularExpressions;
+using WebCrawler.Model;
+
+namespace WebCrawler
+{
+    /// <summary>
+    /// Link parser class.
+    /// </summary>
+    public class LinkParser
+    {
+        #region Constructor
+
+        /// <summary>
+        /// Default constructor.
+        /// </summary>
+        public LinkParser() { }
+
+        #endregion
+        #region Constants
+
+        private const string _LINK_REGEX = "href=\"[a-zA-Z./:&\\d_-]+\"";
+
+        #endregion
+        #region Private Instance Fields
+
+        /// <summary>
+        /// Good Urls
+        /// </summary>
+        private List<string> _goodUrls = new List<string>();
+
+        /// <summary>
+        /// Bad Urls.
+        /// </summary>
+        private List<string> _badUrls = new List<string>();
+
+        /// <summary>
+        /// Other Urls
+        /// </summary>
+        private List<string> _otherUrls = new List<string>();
+
+        /// <summary>
+        /// External Urls
+        /// </summary>
+        private List<string> _externalUrls = new List<string>();
+
+        /// <summary>
+        /// Exceptions
+        /// </summary>
+        private List<string> _exceptions = new List<string>();
+
+        #endregion
+        #region Public Properties
+
+        /// <summary>
+        /// Good Urls.
+        /// </summary>
+        public List<string> GoodUrls
+        {
+            get { return _goodUrls; }
+            set { _goodUrls = value; }
+        }
+
+        /// <summary>
+        /// Bad Urls
+        /// </summary>
+        public List<string> BadUrls
+        {
+            get { return _badUrls; }
+            set { _badUrls = value; }
+        }
+
+        /// <summary>
+        /// Other Urls
+        /// </summary>
+        public List<string> OtherUrls
+        {
+            get { return _otherUrls; }
+            set { _otherUrls = value; }
+        }
+
+        /// <summary>
+        /// External Urls.
+        /// </summary>
+        public List<string> ExternalUrls
+        {
+            get { return _externalUrls; }
+            set { _externalUrls = value; }
+        }
+
+        /// <summary>
+        /// Exceptions
+        /// </summary>
+        public List<string> Exceptions
+        {
+            get { return _exceptions; }
+            set { _exceptions = value; }
+        }
+
+        #endregion
```

Comment:
These can all be [auto implemented properties](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/properties#auto-implemented-properties). Way nicer